Feature/dx 330 adding content pages

W3stside opened the pull request:
Sorry this PR seems so big - it's a lot of removals of dead code and adding static content pages

Just focus on the `claimable` prop being passed in `MenuAuctions` and the selector in `selector/auctions`

Commit:
UD-100 - Partial progress

Commit:
UD-100 - Partial progress

Commit:
UD-110 - Content page init

W3stside committed:
fixed TS error preventing build

W3stside committed:
linting + css fix

Commit:
UD-110 + responsive mobile

Commit:
solid-background for header on contentpage

Commit:
contentpage mods

W3stside committed:
created ContentPage container and component

1. components are all in same file and exported out
2. container renders based on url

W3stside committed:
pass prop to Header to render diff css for ContentPages

1. passed in router.tsx withHeader

W3stside committed:
Generated dynamic routes for ContentPages + Redirects
1. Redirect from anything /content to ;content/HowItWorks
2. redirect to 404 link else broken
3. linting on PageNotFound

W3stside committed:
Moved Footer to own component + changed padding in Header Div

W3stside committed:
added footer to Router under withHeaderAndFooter

W3stside committed:
Added links to Hamburger button via Router

Commit:
fix cherry-pick conflict 1

Commit:
updates jun26 2018

W3stside committed:
fix Hamburger css rebase problem

W3stside committed:
CSS for ClaimButton top of MenuAuctions

1. when any auction is claimable - responsive

W3stside committed:
created ongoingAuctions claimable selector + added in components

1. renders conditional icons based on claimable in menuAuctions

W3stside committed:
added selector in index

W3stside committed:
removed dead unused code

Velenir commented on src/components/Hamburger/index.tsx:
```diff
@@ -21,18 +22,25 @@ export default class Hamburger extends Component<HamburgerProps, HamburgerState>
-          <a href="#">How it works</a>
-          <a href="#">About</a>
-          <a href="#">Faq</a>
+          <Link replace to="/content/HowItWorks"> How It Works </Link>
```
Why replace? It makes me unable to go back to `/` easily

W3stside replied:
i was testing replace and never changed it, thanks

Velenir commented on src/router.tsx:
```diff
@@ -1,40 +1,42 @@
+const withHeaderAndFooter = (Component: React.ComponentClass | React.SFC, content?: boolean) => (props: any) => (
   <Fragment>
-    <Header/>
+    <Header content={content}/>
```
What about instead of a case-specific `content` we pass in `{ headerProps, footerProps }`,  and then destructure them onto `Header, Footer` respectively. Doesn't ultimately matter, of course.

W3stside replied:
/shrug

Velenir commented on src/selectors/auctions.ts:
```diff
@@ -0,0 +1,7 @@
+
+export const auctionClaimable = createSelector(
+  (state: State) => state.auctions.ongoingAuctions,
+  auctions => auctions.find((auc: any) => auc.claim || auc.claimInverse),
```
You a passing in the whole auction instead of a bool. Use `auctions.some(...)`.

W3stside replied:
ah, very cool!

Velenir commented on src/components/Hamburger/index.tsx:
```diff
@@ -21,18 +22,25 @@ export default class Hamburger extends Component<HamburgerProps, HamburgerState>
+          <Link replace to="/content/FAQ">FAQ</Link>
+          <Link replace to="/content/Technical">Technical </Link>
+          <Link replace to="/content/Downtime">Downtime</Link>
+          <Link replace to="/content/Help">Help</Link>
```
You may want to add `onClick={this.handleClick}` to all Links to close the menu. Or not. Depending on the intended UX.

Velenir commented on src/components/ContentPage/index.tsx:
```diff
@@ -0,0 +1,636 @@
+    <section className="drawer">
+      <h3>What do I need to exchange a token on the DutchX?</h3>
+      <p>You only need three things:
+      <ol>
```
Generates `Warning: validateDOMNesting(...): <ol> cannot appear as a descendant of <p>.`. aka _Inside `<p> `phrasing content only, which `<ol>` isn't_ 
A LOT. Console is filled with them. Also in FAQ and Fees

W3stside replied:
mm yeah HTML errors, yay

Velenir commented on src/components/Header/index.tsx:
```diff
@@ -6,8 +6,12 @@ import Hamburger from 'components/Hamburger'
+}
+
+export const Header = ({ content }: HeaderProps) => (
+  <header className={content && 'solid-background'}>
```
Generates `warning.js?1792:33 Warning: Received false for a non-boolean attribute className.` Do a ternary with empty string or use my suggestion about passing `headerProps`

Velenir commented on src/styles/components/navbar/_navbar.scss:
```diff
@@ -15,24 +15,44 @@ header {
 
```
Can you remove `header {width: 100vw }` a few lines above. It adds a horizontal scrollbar when a vertical appears at some sizes.

W3stside replied:
done

Velenir reviewed (changes requested):
With the addition of the `<Footer>` some pages suffer from the _css holy grail_ problem.
Footer appears as soon as it can fit after the main content, which can be easily seen on vertically short pages like `/content/Help`

There are many ways to fix it, many depend on page structure, easiest that comes to mind:

```
#root > div {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
}

footer { margin-top: auto; }
```

Commit:
fixing mobile toggle states

W3stside committed:
removed 'replace' stack for Link

W3stside committed:
changed true && css into ternary + opted for <> over Fragment

W3stside committed:
auctions.find >> auctions.some

W3stside committed:
SCSS changes as discussed with Michel

1. modal centered
2. content pages overflow
3. padding on root > div when smaller

Commit:
fixing misc. styles

W3stside committed:
fixing misc. styles

Commit:
fixing misc. styles

Commit:
fixing misc. styles

W3stside committed:
added onClick behaviour to ContentLinks + Home Button

Commit:
fixing misc. styles

fixing misc. styles

fixing misc. styles

W3stside committed:
ESC key leaves content pages => Home, Behaviour

1. if already on Home page, dont render Home button
2. Clicking ContentPages closes Hamburger button

W3stside committed:
withRouter >> window object && logic fix

1. withRouter switch to window object (if avail)
2. ternary to fix warning.js issue

Velenir reviewed (approved):
Titanic work! approved

W3stside committed:
Merge branch 'develop' into feature/DX-330-Adding-Content-Pages
